typed/validate.go

```diff
@@ -142,7 +142,7 @@ func (v *validatingObjectWalker) doScalar(t schema.Scalar) ValidationErrors {
 }
 
 func (v *validatingObjectWalker) visitListItems(t schema.List, list *value.List) (errs ValidationErrors) {
-	observedKeys := map[string]struct{}{}
+	observedKeys := fieldpath.PathElementSet{}
```

Comment:
Pre-allocating the array to be the right size might save a few allocs. (followup)

Reply:
Clearly, thanks!

Reply:
I finally remembered to do this one, another 20% improvement ;-)